src/Razor/src/Microsoft.CodeAnalysis.Razor.Workspaces/ProjectSystem/ProjectState.cs

```diff
@@ -245,18 +255,10 @@ private ProjectState AddDocument(DocumentState state)
         var documents = Documents.Add(hostDocument.FilePath, state);
 
         // Compute the effect on the import map
-        var importTargetPaths = GetImportDocumentTargetPaths(hostDocument);
-        var importsToRelatedDocuments = AddToImportsToRelatedDocuments(ImportsToRelatedDocuments, hostDocument.FilePath, importTargetPaths);
+        var importsToRelatedDocuments = AddToImportsToRelatedDocuments(hostDocument);
 
-        // Now check if the updated document is an import - it's important this this happens after
-        // updating the imports map.
-        if (importsToRelatedDocuments.TryGetValue(hostDocument.TargetPath, out var relatedDocuments))
-        {
-            foreach (var relatedDocument in relatedDocuments)
-            {
-                documents = documents.SetItem(relatedDocument, documents[relatedDocument].WithImportsChange());
-            }
-        }
+        // Then, if this is an import, update any related documents.
```

Comment:
nit: The check for if it's an import is now in `UpdateRelatedDocuments` and makes this comment awkward for placement. Maybe add as a doc comment on the

Reply:
Actually, this is intentional. This comment goes with the one above to describe the ordering of the algorithm.

```
// Compute the effect on the import map
...

// Then, if this is an import, update any related documents.
...
```

This is in contrast to the comments in `RemoveDocument` below, which are in a different order:

```
// If this is an import, update any related documents.
...

// Then, compute the effect on the import map
...
```

I think your feedback is probably best addressed by renamed the method to something like `UpdateRelatedDocumentsIfNecessary`.